apps/events/models.py

```diff
@@ -847,6 +860,13 @@ def delete(self):
     def is_on_waitlist(self):
         return self in self.event.waitlist_qs
 
+    def is_visible_as_attending(self):
+        # Is visible as attending if default users privacy settings are set and
+        # explicit settings for attending this event does not override privacy setting
+        user_setting = self.user.get_visible_as_attending_events()
+
+        return (user_setting and self.show_as_attending_event is None) or self.show_as_attending_event
```

Comment:
`show_as_attending` sounds like a variable which should be `True` or `False`, not `None`.

Reply:
`None` is used to check if the user has explicitly set a value for this specific event. 

If the value of `show_as_attending_event` is `None` we know that a user has not yet set a preference for this event, and the default value set in the users privacy setting with the button *Synlig på alle sosiale arrangement* is used to find out if the user should be displayed in the list or not. If the value is `False` or `True` we know that the user has specifically set a value for this event.

This way the default value set in the users privacy settings is used by default and if the user wishes be shown/hidden from this specific event the `show_as_attending_event`is changed from `None` to `True` or `False` respectively.

Reply:
I'd think that setting the value to the users specified default _when the user attends_ would achieve the same result.

Attendee.attend() pseudocode:
`attendee.show_as_attending_event = attendee.user.privacy.show_as_attending_events`

This would set `attendee.show_as_attending_event` to the specified default value in the user's privacy settings when the user attends.

This would also make it so that changing the privacy setting does not change the lists for all past events for which you have _not_ opted in to show your attendance.